src/Utils.ts

```diff
@@ -16,17 +16,24 @@ export function fromEditorConfig(
 ): TextEditorOptions {
 	const resolved: TextEditorOptions = {
 		tabSize: (config.indent_style === 'tab'
+			|| config.indent_size === 'tab'
 			? get(config, 'tab_width', config.indent_size)
-			: get(config, 'indent_size', config.tab_width)
-		)
+			: config.indent_style === 'space'
+			? get(config, 'indent_size', config.tab_width)
+			: get(config, 'indent_size', config.tab_width, defaults.tabSize)
+		),
+		insertSpaces: config.indent_style
+			? config.indent_style === 'space'
+			: defaults.insertSpaces
 	};
-	if (get(resolved, 'tabSize') === 'tab') {
-		resolved.tabSize = config.tab_width;
+	if ((!config.indent_style || config.indent_style === 'unset')
```

Comment:
To be clear, there's nothing really special about the word `unset`. It could just as well be `foo` and it should behave the same. The point is that it's not a supported value (i.e., in this case, `tab` or `space`). As such. I'd like to see this read more like:
```suggestion
	if ((config.indent_style !== 'tab' && config.indent_style !== 'space')
```